pkg/userop/middleware.go

```diff
@@ -283,6 +292,68 @@ func getPimlicoERC20PaymasterData(
 	}
 }
 
+func getPimlicoVerifyingPaymsterAndData(
+	bundler RPCBackend,
+	entryPoint common.Address,
+	pimlicoVerifyingConfig PimlicoVerifyingConfig,
+) middleware {
+	return func(ctx context.Context, op *UserOperation) error {
+		opModified := struct {
+			Sender               common.Address `json:"sender"`
+			Nonce                string         `json:"nonce"`
+			InitCode             string         `json:"initCode"`
+			CallData             string         `json:"callData"`
+			CallGasLimit         string         `json:"callGasLimit"`
+			VerificationGasLimit string         `json:"verificationGasLimit"`
+			PreVerificationGas   string         `json:"preVerificationGas"`
+			MaxFeePerGas         string         `json:"maxFeePerGas"`
+			MaxPriorityFeePerGas string         `json:"maxPriorityFeePerGas"`
+			PaymasterAndData     string         `json:"paymasterAndData"`
+			Signature            string         `json:"signature,omitempty"`
+		}{
+			Sender:               op.Sender,
+			Nonce:                fmt.Sprintf("0x%x", op.Nonce.BigInt()),
+			InitCode:             hexutil.Encode(op.InitCode),
+			CallData:             hexutil.Encode(op.CallData),
+			CallGasLimit:         fmt.Sprintf("0x%x", op.CallGasLimit.BigInt()),
+			VerificationGasLimit: fmt.Sprintf("0x%x", op.VerificationGasLimit.BigInt()),
+			PreVerificationGas:   fmt.Sprintf("0x%x", op.PreVerificationGas.BigInt()),
+			MaxFeePerGas:         fmt.Sprintf("0x%x", op.MaxFeePerGas.BigInt()),
+			MaxPriorityFeePerGas: fmt.Sprintf("0x%x", op.MaxPriorityFeePerGas.BigInt()),
+			PaymasterAndData:     hexutil.Encode(op.PaymasterAndData),
+			Signature:            hexutil.Encode(op.Signature),
+		}
+
+		slog.Debug("getting gas limits", "userOp", opModified)
+
+		// Pimlico-standardized gas estimation with paymaster
+		// see https://docs.pimlico.io/paymaster/verifying-paymaster/reference/endpoints#pm_sponsoruseroperation-v2
+		var gasEst gasEstimate
+		if pimlicoVerifyingConfig.SponsorshipPolicyID == "" {
+			if err := bundler.CallContext(ctx, &gasEst, "pm_sponsorUserOperation", opModified, entryPoint); err != nil {
+				return fmt.Errorf("failed to call pm_sponsorUserOperation: %w", err)
+			}
+		} else {
+			if err := bundler.CallContext(ctx, &gasEst, "pm_sponsorUserOperation", opModified, entryPoint, pimlicoVerifyingConfig); err != nil {
+				return fmt.Errorf("failed to call pm_sponsorUserOperation: %w", err)
+			}
+		}
```

Comment:
Just found a way to simplify branching:
```suggestion
    sponsorArgs := []any{opModified, entryPoint}
		if pimlicoVerifyingConfig.SponsorshipPolicyID != "" {
      sponsorArgs = append(sponsorArgs, pimlicoVerifyingConfig)
		}
    if err := bundler.CallContext(ctx, &gasEst, "pm_sponsorUserOperation", sponsorArgs...); err != nil {
      return fmt.Errorf("failed to call pm_sponsorUserOperation: %w", err)
    }
```

Reply:
Oh, nice. Thanks for suggestion!